cmd/dockerd/daemon.go

```diff
@@ -494,10 +502,10 @@ func initRouter(s *apiserver.Server, d *daemon.Daemon, c *cluster.Cluster) {
 	s.InitRouter(debug.IsEnabled(), routers...)
 }
 
-func (cli *DaemonCli) initMiddlewares(s *apiserver.Server, cfg *apiserver.Config) error {
+func (cli *DaemonCli) initMiddlewares(s *apiserver.Server, cfg *apiserver.Config, pluginStore *plugin.Store) error {
 	v := cfg.Version
 
-	exp := middleware.NewExperimentalMiddleware(cli.d.HasExperimental())
+	exp := middleware.NewExperimentalMiddleware(cli.Config.Experimental)
```

Comment:
I'm worried that it might not be so obvious to outside contributors, but I might be overthinking it so I'll leave it up to you.

Knowing that `cli.d` is nil here relies on carefully reading the ordering of code in `start`, which has just changed from previously being the other way around of starting the daemon before initializing middlewares